benchto-driver/src/main/java/io/trino/benchto/driver/execution/QueryExecutionDriver.java

```diff
@@ -61,12 +75,27 @@ private boolean isSelectQuery(String sql)
         return sql.startsWith("select") || sql.startsWith("show") || sql.startsWith("with");
     }
 
-    private QueryExecutionResult executeSelectQuery(Connection connection, QueryExecutionResultBuilder queryExecutionResultBuilder, String sqlStatement)
+    private QueryExecutionResult executeSelectQuery(
+            Connection connection,
+            QueryExecutionResultBuilder queryExecutionResultBuilder,
+            String sqlStatement,
+            Optional<Path> resultFile)
             throws SQLException
     {
         try (Statement statement = connection.createStatement();
                 ResultSet resultSet = statement.executeQuery(sqlStatement)) {
-            int rowsCount = fetchRows(sqlStatement, resultSet);
+            int rowsCount;
+            if (resultFile.isPresent()) {
+                if (Files.exists(resultFile.get())) {
+                    rowsCount = compareRows(resultFile.get(), resultSet);
```

Comment:
add a comment what happens in these 3 if branches. I don't fully understand why we save, compare or fetch rows depending of condition